packages/code-studio/src/settings/ColumnSpecificSectionContent.tsx

```diff
@@ -221,7 +212,7 @@ export class ColumnSpecificSectionContent extends PureComponent<
         let resetKeys = {};
         if (key === 'columnType') {
           resetKeys = {
-            format: '',
+            format: this.makeDefaultFormatterItemByType(value as string),
```

Comment:
I wonder if there's a way using function overloading and/or generics to correct this so you don't need to cast it: https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

Figuring that out isn't really necessary for this though.